src/cart/crt_hg.c

```diff
@@ -327,6 +467,85 @@ crt_hg_addr_lookup(struct crt_hg_context *hg_ctx, const char *name,
 	return rc;
 }
 
+int
+crt_hg_remove_addr (hg_class_t *hg_class, hg_addr_t hg_addr)
+{
+        hg_return_t     ret = HG_SUCCESS;
+
+        D_DEBUG(DB_TRACE, "removing addr from class \n" );
+        ret = HG_Addr_set_remove(hg_class, hg_addr);
+        if (ret != HG_SUCCESS) {
+                D_ERROR("HG_Addr_set_remove) failed, hg_ret %d.\n", ret);
+              	D_GOTO(out, ret = -DER_HG);
+        }
+
+        D_DEBUG(DB_TRACE, "freeing addr from class\n");
+        ret = HG_Addr_free(hg_class, hg_addr);
+        if (ret != HG_SUCCESS) {
+                D_ERROR("HG_Addr_free() failed, hg_ret %d.\n", ret);
+              	D_GOTO(out, ret = -DER_HG);
+        }
+
+out:
+        return ret;
+}
+int
+crt_hg_remove_client_id (uint64_t client_id)
+{
+        hg_return_t     	ret = HG_SUCCESS;
+	d_list_t		*halink;
+	struct crt_ha_mapping 	*ha;
+ 	struct ha_entry 	ha_value;
+
+        halink = d_hash_rec_find(&ha_hash_table,
+                        (void *)&client_id, sizeof(client_id));
+        if (!halink) {
+                D_ERROR("client=%" PRIu64 " not part of the hash table\n", client_id);
+              	D_GOTO(out, ret = -DER_HG);
+        }
+
+        ha = crt_ha_link2ptr(halink);
+        ha_value = ha->ha_value;
+
+        d_hash_rec_decref(&ha_hash_table, halink);
+
+	ret = crt_hg_remove_addr (ha_value.hg_class, ha_value.hg_addr);
+
+        if (ret != HG_SUCCESS) {
+                D_ERROR("crt_remove_addr failed, hg_ret %d.\n", ret);
+              	D_GOTO(out, ret = -DER_HG);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs